src/rezplugins/shell/gitbash.py

```diff
@@ -88,7 +90,43 @@ def get_syspaths(cls):
         cls.syspaths = paths
         return cls.syspaths
 
+    def as_path(self, path):
+        """
+        Return the given path as a system path.
+        Used if the path needs to be reformatted to suit a specific case.
+        Args:
+            path (str): File path.
+
+        Returns:
+            (str): Transformed file path.
+        """
+        return to_posix_path(path)
+
+    def as_shell_path(self, path):
+        """
+        Return the given path as a shell path.
+        Used if the shell requires a different pathing structure.
+
+        Args:
+            path (str): File path.
+
+        Returns:
+            (str): Transformed file path.
+        """
+        return path
```

Comment:
Currently, gets called in `VariantBinding` and by the shell's `setenv`. I'm feeling like when a `{root}` path is processed here it should be posix path and in the case of gitbash with the `shell_pathed_env_vars` setting the way it is, this path here should be converted emulating `cypath -m` so `c:\a\b\c` and `C:\a\b\c` convert to `C:/a/b/c`.

Reply:
This is addressed in the latest commits.